### PR TITLE
Feature: Support consuming files within zip archives

### PR DESCRIPTION
**Pull request by tvhdev**
## Proposed change

Add support for zip files to paperless ngx. E-Mails often contain zipped attachments with documents in it. This code change adds a mechanism which extracts documents from a zip file and processes each file in the consume_file task.

## Type of change

In the consume_file task, a zipfile is extracted and its documents are consumed.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (please explain):

## Checklist:

- [ X] I have read & agree with the [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/main/CONTRIBUTING.md).
- [ ] If applicable, I have included testing coverage for new code in this PR, for [backend](https://docs.paperless-ngx.com/development/#testing) and / or [front-end](https://docs.paperless-ngx.com/development/#testing-and-code-style) changes.
- [ ] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [ ] If applicable, I have checked that all tests pass, see [documentation](https://docs.paperless-ngx.com/development/#back-end-development).
- [X ] I have run all `pre-commit` hooks, see [documentation](https://docs.paperless-ngx.com/development/#code-formatting-with-pre-commit-hooks).
- [ ] I have made corresponding changes to the documentation as needed.
- [ ] I have checked my modifications for any breaking changes.

### CODE REVIEW

**Review by shamoon — CHANGES_REQUESTED** [review]
Thanks. All PRs require testing of added functionality. Something like this would also likely benefit from some kind of documentation as its not totally obvious what uploading a zip file would do.

**Comment by tvhdev** [comment]
@shamoon Thank you for your feedback. I will try to write testcase for the new functionality. The change in the consume_file task extracts all files of a given zip archive and cosumes each file. the zipfile itself is not stored.

**Comment by tvhdev** [comment]
@shamoon Hello, i added a testcase to test the upload of a zipfile.

**Comment by stumpylog** [comment]
I'm unsubscribing until this is ready.  Ping me when it is

**Comment by shamoon** [comment]
If you'd like to keep working on this until it's actually ready please close it and re-open. But I dont think its ready as-is.

File: Pipfile.lock
Comment (shamoon) [inline Pipfile.lock:1]: Please dont remove the pipenv lock file

**Comment by shamoon** [comment]
Please finish up the PR in your own fork and re-open when its ready

**Pull request closed without merging**